### PR TITLE
Optimize collision

### LINKED ISSUE #56: Isaac direction
Isaac should retain its direction even when he stoops.

### PR DESCRIPTION
**Pull request by Thomyrock**
Fix #56 #52 #50 #49

### CODE REVIEW

File: src/init/global.init.s [hunk 1]
```diff
@@ -4,20 +4,34 @@
...
+; ////// ROCK \\\\\\
 	ld a,%10010000	; size = 0, block, hurt by bombs.
 	ldi (hl),a
 	ld a,0	; dmg = 0
 	ldi (hl),a
 	ld a,0
 	ldi (hl),a
 	ldi (hl),a ; no funtion pointer or whatever
+	ldi (hl),a ; speed = 0
+	ld a,0
```
Comment (DorianXGH) [inline src/init/global.init.s:17]: Why ? a is already 0

File: src/init/global.init.s [hunk 2]
```diff
@@ -4,20 +4,34 @@
...
+	ldi (hl),a ; 0 HP
+; \\\\\\ ROCK //////
+
+; ////// VOID \\\\\\
+	ld a,%00010000	; size = 0, block, hurt by bombs.
+	ldi (hl),a
+	ld a,0	; dmg = 0
+	ldi (hl),a
+	ld a,0
```
Comment (DorianXGH) [inline src/init/global.init.s:26]: same

File: src/init/global.init.s [hunk 3]
```diff
@@ -4,20 +4,34 @@
 global_init:
 	ld hl,global_.sheets
 	ld b,n_sheets
-@sheets_loop:
-;	for now they are all rocks
+
+; ////// ROCK \\\\\\
 	ld a,%10010000	; size = 0, block, hurt by bombs.
 	ldi (hl),a
 	ld a,0	; dmg = 0
```
Comment (DorianXGH) [inline src/init/global.init.s:11]: xor a -> faster

File: src/init/global.init.s [hunk 4]
```diff
@@ -44,13 +58,13 @@ global_init:
 	ld de,global_.states
 @elements_loop: ; they are no element for now
 	xor a
+	ldi (hl),a ; hp = 0
 	ldi (hl),a ; x = 0
 	ldi (hl),a ; y = 0
-	ldi (hl),a ; speed x = 0, speed y = 0
 	ldi (hl),a ; sheet = 0
 	ld a,0
```
Comment (DorianXGH) [inline src/init/global.init.s:65]: same : a is already 0

File: src/movement_and_collide.s [hunk 5]
```diff
@@ -1,146 +1,155 @@
...
 
-; //////// Collision \\\\\\\
+;   //// collision X  init \\\\
 	ld a, (global_.isaac.x)
 	ld (collision_.p.1.x), a
-	ld a, (global_.isaac.y)
-	ld (collision_.p.1.y), a
-	ld a, ISAACHITBOX
-	ld (collision_.hitbox1), a
+;   \\\\ collision Y  init ////
```
Comment (DorianXGH) [inline src/movement_and_collide.s:97]: Isn't it X ?

File: src/init/global.init.s (hunk 1 above)
Comment (JeanMarieMineau) [reply src/init/global.init.s:17]: It's easier to modify (it the element HP, set it to 1 add the element in dysplay_init.s)
It's an init file, performance isn't an issue.

File: src/init/global.init.s (hunk 2 above)
Comment (JeanMarieMineau) [reply src/init/global.init.s:26]: same, it just happens to be 0, but we may want to change it.

File: src/init/global.init.s (hunk 3 above)
Comment (JeanMarieMineau) [reply src/init/global.init.s:11]: same

File: src/init/global.init.s (hunk 4 above)
Comment (JeanMarieMineau) [reply src/init/global.init.s:65]: same

File: src/movement_and_collide.s (hunk 5 above)
Comment (JeanMarieMineau) [reply src/movement_and_collide.s]: Yes, y just corrected it.

**Pull request merged**